src/executioners/SpectralExecutionerLinearElastic.C

```diff
@@ -0,0 +1,402 @@
+                                                    const RankFourTensor & elasticity_tensor)
+{
+  const auto & grid = gamma_hat.grid();
+  const int ndim = 3;
+
+  const int ni = grid[0];
+  const int nj = grid[1];
+  const int nk = (grid[2] >> 1) + 1;
+
+  std::size_t index = 0;
+
+  const auto & ivec = gamma_hat.kTable(0);
+  const auto & jvec = gamma_hat.kTable(1);
+  const auto & kvec = gamma_hat.kTable(2);
+
+  const std::vector<int> grid_vector = gamma_hat.grid();
+
+  const Complex I(0.0, 1.0);
+
+  // Fill fourth-order Green operator based on homogeneous material properties
+  for (int freq_x = 0; freq_x < ni; ++freq_x)
+    for (int freq_y = 0; freq_y < nj; ++freq_y)
+      for (int freq_z = 0; freq_z < nk; ++freq_z)
+      {
+        const std::array<Complex, 3> freq{ivec[freq_x] * I, jvec[freq_y] * I, kvec[freq_z] * I};
+
+        Real lambda0 = _young_modulus * _average_factor * _poisson_ratio /
+                       ((1 + _poisson_ratio) * (1 - 2 * _poisson_ratio));
+        Real nu0 = _young_modulus * _average_factor / (2 * (1 + _poisson_ratio));
+        Real constant = (lambda0 + nu0) / (nu0 * (lambda0 + 2.0 * nu0));
+
+        for (int i = 0; i < ndim; i++)
+          for (int j = 0; j < ndim; j++)
+            for (int k = 0; k < ndim; k++)
+              for (int l = 0; l < ndim; l++)
+              {
+                Complex q_square = freq[0] * freq[0] + freq[1] * freq[1] + freq[2] * freq[2];
+                if (std::abs(q_square) > 1.0e-12)
+                {
+                  gamma_hat.reciprocalSpace()[index](i, j, k, l) =
+                      -1.0 * constant * (freq[i] * freq[j] * freq[k] * freq[l]) /
+                          (q_square * q_square) +
+                      (deltaij(i, k) * freq[j] * freq[l] + deltaij(j, k) * freq[i] * freq[l] +
+                       deltaij(i, l) * freq[j] * freq[k] + deltaij(j, l) * freq[i] * freq[k]) /
+                          (4.0 * nu0 * q_square);
+                }
+                else
+                {
+                  gamma_hat.reciprocalSpace()[index] = elasticity_tensor.invSymm();
+                }
+              }
+
+        index++;
+      }
+}
+
+FFTBufferBase<RankTwoTensor> &
+SpectralExecutionerLinearElastic::getInitialStress(
+    FFTBufferBase<RankTwoTensor> & epsilon_buffer,
+    const FFTBufferBase<RankFourTensor> & elastic_tensor)
+{
+  auto & stress_buffer = getFFTBuffer<RankTwoTensor>("stress");
+  const auto & grid = epsilon_buffer.grid();
+  int ni = grid[0];
+  int nj = grid[1];
+  int nk = grid[2];
+
+  // Homogeneous initial state of strain
+  fillOutEpsilonBuffer(epsilon_buffer);
+
+  size_t index = 0;
+  for (int freq_x = 0; freq_x < ni; ++freq_x)
+    for (int freq_y = 0; freq_y < nj; ++freq_y)
+      for (int freq_z = 0; freq_z < nk; ++freq_z)
+      {
+        stress_buffer.realSpace()[index] =
+            elastic_tensor.realSpace()[index] * epsilon_buffer.realSpace()[index];
+        index++;
+      }
```

Comment:
We should have a templated helper function to do these types multiplications in `FFTData`

```
template <typename T>
template <typename T1, typename T2>
void FFTData<T>::multiply(const FFTData<T1> & m1, const FFTData<T1> & m2)
{
  // assert that grid sizes are identical
  // loop over indices and multiply
}
```

Reply:
Grid size information is in `FFTBuffer`. In `FFTData` you have either a real or a complex buffer. I am adding a version of your suggestion.